src/plugins/index_pattern_field_editor/public/open_delete_modal.tsx

```diff
@@ -40,6 +40,16 @@ export const getFieldDeleteModalOpener = ({
   indexPatternService,
   usageCollection,
 }: Dependencies) => (options: OpenFieldDeleteModalOptions): CloseEditor => {
+  if (typeof options.fieldName === 'string') {
```

Comment:
For the demo we prevent deleting a subField directly as this will have to be done by editing the runtime object script.